# Project: 0x05. Python - Exceptions

<h2>Learning Objectives</h2>

<h3>General</h3>

<ul>
<li>Why Python programming is awesome</li>
<li>What’s the difference between errors and exceptions</li>
<li>What are exceptions and how to use them</li>
<li>When do we need to use exceptions</li>
<li>How to correctly handle an exception</li>
<li>What’s the purpose of catching exceptions</li>
<li>How to raise a builtin exception</li>
<li>When do we need to implement a clean-up action after an exception</li>
</ul>

<h2>Tasks</h2>

| Task | File |
| ---- | ---- |
| 0. Safe list printing | [0-safe_print_list.py](./0-safe_print_list.py) |
| 1. Safe printing of an integers list | [1-safe_print_integer.py](./1-safe_print_integer.py) |
| 2. Print and count integers | [2-safe_print_list_integers.py](./2-safe_print_list_integers.py) |
| 3. Integers division with debug | [3-safe_print_division.py](./3-safe_print_division.py) |
| 4. Divide a list | [4-list_division.py](./4-list_division.py) |
| 5. Raise exception | [5-raise_exception.py](./5-raise_exception.py) |
| 6. Raise a message | [6-raise_exception_msg.py](./6-raise_exception_msg.py) |

